objurl/objurl.go

```diff
@@ -209,6 +212,9 @@ func (o *ObjectURL) setPrefixAndFilter() error {
 		o.Prefix = o.Path[:loc]
 		o.filter = o.Path[loc:]
 	}
+	for _, char := range regexCharacters {
+		o.Prefix = strings.Replace(o.Prefix, string(char), "\\" + string(char), -1)
```

Comment:
I'm not sure why this change is necessary. We call `regexp.QuoteMeta` a couple of lines below for escaping. Could you demonstrate a use-case please?

Reply:
The calls to `regexp.QuoteMeta` below only escape characters in the `filter`. In the case of a source URL such as the one given in the new testcase, regex metacharacters will be found in the `prefix` as well. In the case of that testcase, an error will be thrown since it's a malformed regex.

Also, thanks for mentioning `regexp.QuoteMeta`. I'm new to `golang` and I didn't realize this existed.